### PR TITLE
CVE 2011-1113 and CVE 2010-1504

### PR DESCRIPTION
**Pull request by roda0313**
Adding information for CVE 2011-1113 and CVE 2010-1504

### CODE REVIEW

File: cves/CVE-2010-1504.yml [hunk 1]
```diff
@@ -32,7 +32,10 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  When downloading a file in chrome, the downloaded file could intiate a cross site scripting attack because
+  when the download url was passed to the javascript code, it was directly placing the url into the html
+  of the page, allowing for easy script execution
```
Comment (kbaumzie) [inline cves/CVE-2010-1504.yml]: Each description should be about two paragraphs; i would focus more on the vulnerability, its introduction, severity, and its fix

File: cves/CVE-2010-1504.yml [hunk 2]
```diff
@@ -197,4 +213,11 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    This was a design mistake that led to this bug. In this intial code, it was not taken into account that
+    a user may provide a bad link with html code directly in it. This bug is also so old, that it is likely that it
+    was unknown exactly what kinds of attacks were possible at the time. The fix for this was correct but not complete
+    and later, a complete solution was added essentially making it impossible to use an XSS attack nowadays using the
```
Comment (RobertHofmann) [inline cves/CVE-2010-1504.yml]: If an XSS attack is only 'essentially impossible', is it still possible?

File: cves/CVE-2011-1113.yml [hunk 3]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2011-1113
```
Comment (RobertHofmann) [inline cves/CVE-2011-1113.yml]: Same as before, this should be a CWE number, not a link to a CVE.

File: cves/CVE-2011-1113.yml [hunk 4]
```diff
@@ -32,7 +32,11 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Chrome uses a byte stream deserializer called a pickler. The pickler function FindNext has
```
Comment (RobertHofmann) [inline cves/CVE-2011-1113.yml]: The description should not have Chromium specific stuff.

File: cves/CVE-2010-1504.yml [hunk 5]
```diff
@@ -46,16 +49,16 @@ reviews:
 - 1593007
 bugs:
 - 40138
-repo: 
+repo: https://bugs.chromium.org/p/chromium/issues/detail?id=40138
 fixes_vcc_instructions: |
   Please put the commit hash in "commit" below (see my example in
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
 - :commit: 7eb546401ee80432c3a7498300168424909bedf8
```
Comment (RobertHofmann) [inline cves/CVE-2010-1504.yml]: In your conclusion, you mention a partial and a complete fix. Perhaps both of these should be listed here.

File: cves/CVE-2011-1113.yml [hunk 6]
```diff
@@ -196,7 +220,11 @@ mistakes:
     Look at the CWE entry for this vulnerability and examine the mitigations
     they have written there. Are they doing those? Does the fix look proper?
 
-    Use those questions to inspire your answer. Don't feel obligated to answer
+    Use those questiCVE-2010-1504ons to inspire your answer. Don't feel obligated to answer
```
Comment (RobertHofmann) [inline cves/CVE-2011-1113.yml]: You accidentally pasted the CVE number here. Should be removed.

File: cves/CVE-2011-1113.yml [hunk 7]
```diff
@@ -196,7 +220,11 @@ mistakes:
...
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    This was a very simple mistake that developers make all the time. The original developer assumed all input to
+    this class would be good and no validation/sanitation was required. In any instance where memory is directly being
+    affected, buffer sizes must always be checked and anything larger than the maximum buffer size must be
+    disallowed
```
Comment (RobertHofmann) [inline cves/CVE-2011-1113.yml]: 3 upvotes for this one

File: cves/CVE-2010-1504.yml [hunk 8]
```diff
@@ -197,4 +213,11 @@ mistakes:
...
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    This was a design mistake that led to this bug. In this intial code, it was not taken into account that
+    a user may provide a bad link with html code directly in it. This bug is also so old, that it is likely that it
+    was unknown exactly what kinds of attacks were possible at the time. The fix for this was correct but not complete
+    and later, a complete solution was added essentially making it impossible to use an XSS attack nowadays using the
+    same attack as before. The original developer also perhaps did not understand that the line of code
+    "url.InnerHtml = url" would set the actual web pages executable html to whatever the
+    user inputs, which is of course bad.
```
Comment (RobertHofmann) [inline cves/CVE-2010-1504.yml]: 2 upvotes for this one

File: cves/CVE-2010-1504.yml (hunk 2 above)
Comment (roda0313) [reply cves/CVE-2010-1504.yml]: It is not possible using the same method was was used before. I never like to say never when it comes to security but I can not think of a way this attack could be a problem again given the existing code base.

File: cves/CVE-2010-1504.yml [hunk 9]
```diff
@@ -32,7 +32,13 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  This problem was found early on in chromes development and for good reason. If you passed an html <script> tag to
```
Comment (RiceOnly) [inline cves/CVE-2010-1504.yml]: chrome's

File: cves/CVE-2010-1504.yml [hunk 10]
```diff
@@ -94,20 +102,24 @@ discovered:
...
   question: |
     What subsystems was the mistake in?
 
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: 
-  name: 
+  answer: |
+    This is located under the chrome browser resources subsystem and is literally just part of the script that
```
Comment (RiceOnly) [inline cves/CVE-2010-1504.yml]: use of 'literally' is unnecessary

File: cves/CVE-2010-1504.yml [hunk 11]
```diff
@@ -117,10 +129,14 @@ interesting_commits:
     emerging themes?
 
     If there are no interesting commits, demonstrate that you completed this section by explaining what happened between the VCCs and the fix.
-  answer:
+  answer: |
+    Between the VCC and the fix commit there is not really any interesting commits, however, after the initial fix there
+    have been a number of vast security improvements to make this even more secure and dissallow XSS attacks.
+    The file in todays chrome repo, has been moved to chrome/browser/resources/downloads/downloads.js and there are a
```
Comment (RiceOnly) [inline cves/CVE-2010-1504.yml]: they instead of there

File: cves/CVE-2010-1504.yml [hunk 12]
```diff
@@ -197,4 +216,11 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    This was a design mistake that led to this bug. In this intial code, it was not taken into account that
```
Comment (RiceOnly) [inline cves/CVE-2010-1504.yml]: In the initial code

File: cves/CVE-2010-1504.yml [hunk 13]
```diff
@@ -197,4 +216,11 @@ mistakes:
...
+    a user may provide a bad link with html code directly in it. This bug is also so old, that it is likely that it
+    was unknown exactly what kinds of attacks were possible at the time. The fix for this was correct but not complete
+    and later, a complete solution was added essentially making it impossible to use an XSS attack nowadays using the
+    same attack as before. The original developer also perhaps did not understand that the line of code
+    "url.InnerHtml = url" would set the actual web pages executable html to whatever the
```
Comment (RiceOnly) [inline cves/CVE-2010-1504.yml]: 3 upvotes

File: cves/CVE-2011-1113.yml [hunk 14]
```diff
@@ -78,9 +82,11 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    Yes, there were automation tests created to ensure the fix for this bug was correctly implemented
+    as well as check that it returns and error if the header size and actual size input does not match
```
Comment (RiceOnly) [inline cves/CVE-2011-1113.yml]: an error instead of and error

File: cves/CVE-2011-1113.yml [hunk 15]
```diff
@@ -119,12 +132,16 @@ interesting_commits:
     emerging themes?
 
     If there are no interesting commits, demonstrate that you completed this section by explaining what happened between the VCCs and the fix.
-  answer:
+  answer: |
+    There really are no interesting commits between the time this file was originally added and the fix commit.
+    Since this file is such a key, critical file to the entire system, it was implemented and developed very early on and
+    it has barely been touched since its original implementation. This file was originall added in commit number 8 and
```
Comment (RiceOnly) [inline cves/CVE-2011-1113.yml]: originally

File: cves/CVE-2011-1113.yml [hunk 16]
```diff
@@ -133,7 +150,10 @@ major_events:
 
     The event doesn't need to be directly related to this vulnerability, rather,
     we want to capture what the development team was dealing with at the time.
-  answer: 
+  answer: |
+    There are no major events. The bug was found rather early in development and about 3 years after
+    chromium was intially launched. Pickle.cc has no real dependancies on the rest of the system, but many subsystems
```
Comment (RiceOnly) [inline cves/CVE-2011-1113.yml]: dependencies from dependancies

File: cves/CVE-2011-1113.yml [hunk 17]
```diff
@@ -199,4 +223,8 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    This was a very simple mistake that developers make all the time. The original developer assumed all input to
+    this class would be good and no validation/sanitation was required. In any instance where memory is directly being
+    affected, buffer sizes must always be checked and anything larger than the maximum buffer size must be
```
Comment (RiceOnly) [inline cves/CVE-2011-1113.yml]: upvote 2

**Pull request merged**